src/BenchmarkDotNet/Running/BenchmarkRunnerClean.cs

```diff
@@ -390,8 +380,19 @@ private static ImmutableArray<ValidationError> Validate(params BenchmarkRunInfo[
 
             var buildResults = buildPartitions
                 .AsParallel()
-                .Select(buildPartition => (buildPartition, buildResult: Build(buildPartition, rootArtifactsFolderPath, buildLogger)))
-                .ToDictionary(result => result.buildPartition, result => result.buildResult);
+                .ToDictionary(
+                    buildPartition => buildPartition,
+                    buildPartition =>
+                    {
+                        var buildResult = Build(buildPartition, rootArtifactsFolderPath, buildLogger);
+
+                        // If the generation was successful, but the build was not, we will try building sequentially
+                        // so don't send the OnBuildComplete event yet.
+                        if (buildPartitions.Length <= 1 || !buildResult.IsGenerateSuccess || buildResult.IsBuildSuccess)
+                            eventProcessor.OnBuildComplete(buildPartition, buildResult);
```

Comment:
It looks like this is being called concurrently, which necessitates synchronization. Perhaps it would be better to take the parallel query and force it back to sequential to invoke on the same thread?

```cs
buildResults = buildPartitions
    .AsParallel()
    .Select(buildPartition => (buildPartition, buildResult: Build(buildPartition, rootArtifactsFolderPath, buildLogger)))
    .AsSequential()
    .ToDictionary(...)
```

Reply:
Nice suggestion, I did not know that `AsSequential` existed. Adding that in now.